firmware/Core/Src/telecommands/telecommand_executor.c

```diff
@@ -173,3 +173,33 @@ uint8_t TCMD_execute_telecommand_in_agenda(const uint16_t tcmd_agenda_slot_num,
         response_output_buf_size
     );
 }
+
+/// @brief Deletes all entries from the agenda.
+/// @return 0 on completion.
+uint8_t TCMD_agenda_delete_all() {
+    for (uint16_t slot_num = 0; slot_num < TCMD_AGENDA_SIZE; slot_num++) {
+        if (TCMD_agenda_is_valid[slot_num]) {
+            TCMD_agenda_is_valid[slot_num] = 0;
+        }
+    }
+    return 0;
+}
+
+/// @brief Deletes a telecommand from the agenda by its `timestamp_sent` field.
+/// @param tssent The `timestamp_sent` value of the telecommand to delete.
+/// @return 0 on success, 1 if the telecommand was not found.
+uint8_t TCMD_agenda_delete_by_tssent(uint64_t *tssent) {
```

Comment:
Just noticed it, but can you please change the `tssent` arg to be a regular into instead of a pointer.

Reply:
Done, I removed the pointer tag from the function and the function definition and passed the `const uint8_t tssent` variable into the function instead of a pointer.

Is there a reason I should avoid using a pointer for that instant?

Reply:
Using a pointer is unnecessary, allows for the introduction of accidental errors. Additionally, it prevents the value from being inadvertently modified from within the function.

Reply:
Thanks that makes sense